### PR TITLE
Buy Sell Stock K Items implemented in cpp

### PR DESCRIPTION
**Pull request by saketbyte**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes I have gone through the documentation.

### Description
Wrote a program as proposed in #4407 with self-explicable comments to understand each line and execution thoroughly and have checked on multiple test cases.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Related to #4407 . Kindly merge if it is up to the required standards or suggest changes if any,

### CODE REVIEW

File: C-Plus-Plus/dp/buy_sell_stock_k_time.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+                    max = prev_profit + curr_profit;
+            }
+
+            dp[transaction][day] = max;
+        }
+    }
+
+    cout << dp[k][n - 1];
+
+}
```
Comment (gat786) [inline C-Plus-Plus/dp/buy_sell_stock_k_time.cpp]: Add a line break {i.e. new line} here

File: C-Plus-Plus/dp/buy_sell_stock_k_time.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
+/*
```
Comment (gat786) [inline C-Plus-Plus/dp/buy_sell_stock_k_time.cpp:1]: First of all your title for this pr is `Mybranch` make it something relevant to the code you are writing. 
A suggested different title would be `Buy Sell Stock K Items implemented in cpp`. It just adds so much more context to the pr

**Review by gat786 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/dp/buy_sell_stock_k_time.cpp (hunk 2 above)
Comment (gat786) [reply C-Plus-Plus/dp/buy_sell_stock_k_time.cpp:1]: nvm i took care of it

**Comment by saketbyte** [comment]
Thank you, I shall update accordingly. My apologies for inappropriate pull request name. I might have mistaken it for branch name.

**Comment by saketbyte** [comment]
I updated the changes requested, kindly update if there are further changes needed or please merge the pull request, it has been 3 days.

File: C-Plus-Plus/dp/buy_sell_stock_k_time.cpp [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+    int n, k;
+    cin >> n >> k; // taking inputs for number of days and the number of transactions.
+
+    int P[n];
+
+    for (int i = 0; i < n; i++)
+        cin >> P[i];
+
+<<<<<<< HEAD
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/buy_sell_stock_k_time.cpp]: Seems like these should not be here

File: C-Plus-Plus/dp/buy_sell_stock_k_time.cpp [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    int n, k;
+    cin >> n >> k; // taking inputs for number of days and the number of transactions.
+
+    int P[n];
+
+    for (int i = 0; i < n; i++)
+        cin >> P[i];
+
+=======
+>>>>>>> c733b13aa7e4728a5b78f40398b46fb05ceb71ca
```
Comment (HarshCasper) [inline C-Plus-Plus/dp/buy_sell_stock_k_time.cpp]: Seems like these should not be here

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Can you remove extra lines of code as well?

File: C-Plus-Plus/dp/buy_sell_stock_k_time.cpp (hunk 3 above)
Comment (gat786) [reply C-Plus-Plus/dp/buy_sell_stock_k_time.cpp]: Yeah these aren't part of actual code

File: C-Plus-Plus/dp/buy_sell_stock_k_time.cpp (hunk 4 above)
Comment (gat786) [reply C-Plus-Plus/dp/buy_sell_stock_k_time.cpp]: This seems to be a git conflict residue

**Comment by gat786** [comment]
Do the changes as pointed out by Harsh.

**Review by ankitaggarwal23 — COMMENTED** [review]
Are you still working on it.

File: C-Plus-Plus/dp/buy_sell_stock_k_time.cpp [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+ * INPUT:
+ * int n - the number of days for which transactions are to be made
+ * int k - the number of non-overlapping transactions to be made
+ * int A[n] - array containing the price for each day of the stock
+ *
+ * OUTPUT: Output a single integer which is the maximum profit made if allowed k transactions at most.
+ *
+ */
+
+#include <iostream>
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/buy_sell_stock_k_time.cpp:14]: Remove the iostream header

File: C-Plus-Plus/dp/buy_sell_stock_k_time.cpp [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    *          Now any (i,j) element in the above matrix represents the best profit made until ith day using j transactions.
+    *          The (i,j) result will depend upon the (i-1,j) cell and all the cell of j-1 column before i ie in j-1 from 0 to i-1,
+    *          because we can either make the final transaction on (i,j) after previous day (i-1,j)'s best transaction or after
+    *          the k-1th transaction on any i-1'th day.
+    *          So we compare the buying and selling point of the above and check for the best profit made till today.
+    *
+    *          We run nested for loops to implement above logic. Here we keep track of maximum profit made till today and
+    *          maximum profit made on the same day, and we store only the maximum profit made till today in our dp[][].
+    *
+    */
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/dp/buy_sell_stock_k_time.cpp:49]: Remove the part

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add Multi-Line Comments 
- Add a beginner-friendly description at the top
- Use proper code Format 
- Remove extra lines of code and spaces
- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

**Review by ankitaggarwal23 — COMMENTED** [review]
Last Reminder, Are you still working on it.

**Review by ankitaggarwal23 — COMMENTED** [review]
Closing this due to inactivity. Thank you for your valuable time.😊

**Pull request closed without merging**